server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryParseRequest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportRequest;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: request to extensions to parse context
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryParseRequest extends TransportRequest {
+
+    private final Class categoryClass;
+    private byte[] context;
+
+    /**
+     * @param categoryClass Class category for this parse request
+     * @param context StreamInput object to convert into a byte array and transport to the extension
+     * @throws IllegalArgumentException if context bytes could not be read
+     */
+    public NamedWriteableRegistryParseRequest(Class categoryClass, StreamInput context) {
+        try {
```

Comment:
Do we need a try/catch block here because we are just assigning values in the constructor?